File: DSA/Sauhard_Semalty_590013630/Day9Question1.py
import sys

def is_palindrome(string: str) -> bool:
    "Checks if the string is palindrome (case insensitive)."

    string = string.lower()
    half = len(string) // 2

    for i in range(half):
        if string[i] != string[-1 - i]:
            return False

    return True


def main() -> None:
    "The entry point of the program."

    string = input("Enter string: ")
    if not string:
        print("Empty string is not allowed.")
        sys.exit(-1)

    # Check for palindrome
    print(str(is_palindrome(string)).lower())


if __name__ == "__main__":
    main()

File: DSA/Stuti_Jain_590012149/Day_9_Question_1.c
#include<stdio.h>
#include<stdbool.h>
char s[100];
int len, i;
int main(){
    printf("Enter the string: ");
    scanf("%s", &s);
    
     for(i = 0; s[i]; i++) {
        len++;
    }


    bool Palindrome=true;

    for(i=0; i<len/2; i++){
        if(s[i]!=s[len-i-1]){
            Palindrome=false;
            break;
        }
    }

    if(Palindrome){
        printf("true\n");
    } 
    else {
        printf("false\n");
    }

    return 0;
}
